src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -684,6 +783,13 @@ public TypeHandle GetArrayElementTypeHandle()
 
         [MethodImpl(MethodImplOptions.InternalCall)]
         public extern uint GetNumInstanceFieldBytes();
+
+        /// <summary>
+        /// Get the <see cref="CorElementType"/> representing this type. Enums are represented by underlying type.
+        /// </summary>
+        /// <remarks>This method is optimized in favor of primitive types and enums.</remarks>
```

Comment:
The current implementation of the method can return wrong result for non-primitive/non-enum types. For example, it will return `ELEMENT_TYPE_I` for `RuntimeArgumentHandle`

Reply:
Since it would be used in performance sensitive case (deciding conversion between `int[]`<->`uint[]`), I think it's better to add assert and rename the method, rather than changing the implementation.

Reply:
Sounds fine to me.